tests/CMakeLists.txt

```diff
@@ -186,9 +186,12 @@ add_test(NAME test_log_negative
 set_tests_properties(test_log_negative
                      PROPERTIES PASS_REGULAR_EXPRESSION "Invalid log level")
 
+if (BUILD_OSTREE)
+    set(TEST_IPSEC_ARGS "--ostree")
+endif ()
 add_test(NAME test_ip_secondary
          COMMAND ${PROJECT_SOURCE_DIR}/tests/ipsecondary_test.py
-         --build-dir ${PROJECT_BINARY_DIR} --src-dir ${PROJECT_SOURCE_DIR} --ostree ${BUILD_OSTREE})
+         --build-dir ${PROJECT_BINARY_DIR} --src-dir ${PROJECT_SOURCE_DIR} ${TEST_IPSEC_ARGS})
```

Comment:
What `${TEST_IPSEC_ARGS}` is for?

Reply:
It's here to set or not set `--ostree` as an argument to the script, as using `${BUILD_OSTREE}` in the python script would force us to reimplement CMake boolean logic:

> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. If the argument is not one of these specific constants, it is treated as a variable or string and the following signature is used.
https://cmake.org/cmake/help/latest/command/if.html#command:if

I had the problem that the tests would not run because I configured my build directory with `-DBUILD_OSTREE=on` instead of `-DBUILD_OSTREE=ON`.

Reply:
OK, I got it.